### PR TITLE
Remove OPCODE_LINE from GDScript

### PR DESCRIPTION
**Pull request by bojidar-bg**
Additionally, refactor `ScriptDebugger::breakpoints` a bit. Also, make `OPCODE` macros more similar between the switch and jump table versions.

I think my implementation still has a few bugs and rough spots, but I'm happy to announce that it works without crashing or getting into endless loops just fine.

This PR is mostly a PoC of something I was thinking of for some time now, but if deemed useful, I could try to fix the remaining issues (most of which are one-by-off errors here and there).

### CODE REVIEW

**Comment by Calinou** [comment]
Does this provide a performance improvement? Just curious about it :slightly_smiling_face:

**Comment by akien-mga** [comment]
I'm also curious about what's the consequence of this change in layman's terms? I guess it might be clear for @vnen / @reduz but the rest of us need to be hyped too :P

**Comment by bojidar-bg** [comment]
So far, I haven't gotten around to testing the performance of the changed code, though I think it might be a bit slower due to the way `_get_ip_line` is currently implemented.

Anyway, here is a list of reasons we might want the change:

* `OPCODE_LINE` was generated only in debug builds. If we want to export the generated opcodes instead of just a tokenized version of the script, we would have to recompile an "export" version without `OPCODE_LINE`.
* `OPCODE_LINE` slows down every line in debug runs, even when there are no breakpoints for the script (it requires an additional jump every line, and gets to call ScriptDebugger two or three times). On the other hand, this solution adds a `unlikely` jump to every opcode, and a (hopefully) not-too-slow piece of code which checks for breakpoints on every jump.
* Currently, this PR makes it so that line numbers can be dereferenced even in release builds. Previously, (AFAIS) line numbers would be `0` in GDScript stack traces of release builds.
* Changing the way breakpoints are handled happens to resolve a few issues with breakpoints.

And a few reasons we might *not* want it:

* Change for the sake of change is not good; there has to be some demonstrable benefit (past the few code refactorings).
* It might happen to be slower than the previous version. I don't know for sure yet.
* It introduces a few issues with line numbers and stepping, which I haven't resolved yet.
* It is partly permutation programming, due to the large amount of off-by-one mishaps possible.

**Comment by aaronfranke** [comment]
@bojidar-bg Is this still desired? If so, it needs to be rebased on the latest master branch.

@vnen Is this PR still relevant with the GDScript rewrite?

**Comment by bojidar-bg** [comment]
Going to close + salvageable this; I will revisit the topic after GDScript 2.0, in case it was not implemented that way there.
